pkg/apis/apps/v1alpha1/statefulset_types.go

```diff
@@ -23,6 +24,32 @@ import (
 	"k8s.io/apimachinery/pkg/util/intstr"
 )
 
+const (
+	// StatefulSetInPlaceUpdateReady must add into template.spec.readinessGates when pod podUpdatePolicy
+	// is InPlaceIfPossible or InPlaceOnly. The condition in podStatus will be updated to False before in-place
+	// updating and updated to True after finished updating. This ensures pod being not-ready during
+	// in-place updating.
+	StatefulSetInPlaceUpdateReady v1.PodConditionType = "InPlaceUpdateReady"
```

Comment:
Instead of forcing user to set the readiness gate in the statefulset's template,  can this be done by the kruise-controller to update the generated pod spec to include the readiness gate

Reply:
I don't think it is a good idea to update readinessGates automatically in controller or webhook.. Modification of statefulset template should be very careful because it could raise pod updating.

For example, someone create a statefulset with 100 replicas, then he updates podUpdatePolicy to `InPlaceIfPossible`. If kruise-manager find this and add `InPlaceUpdateReady` into readinessGates, then pods start to be deleted because of recreate update. This could make user confused and is unsafe for online applications.

Reply:
I meant the controller do this when creating the pods.
It's a good question what if user wants to update the policy to `InPlaceIfPossible` regardless of how readiness is specified originally, in this case the `readinessGate` cannot be updated to the existing pods.